pkg/engine/imageVerify.go

```diff
@@ -153,15 +153,38 @@ func (iv *imageVerifier) verify(imageVerify *v1.ImageVerification, images map[st
 				continue
 			}
 
+			if imageVerify.DigestMutate == nil {
+				digest := true
+				imageVerify.DigestMutate = &digest
+			}
+
 			var ruleResp *response.RuleResponse
 			if len(imageVerify.Attestations) == 0 {
 				var digest string
 				ruleResp, digest = iv.verifySignature(imageVerify, imageInfo)
-				if ruleResp.Status == response.RuleStatusPass {
-					iv.patchDigest(path, imageInfo, digest, ruleResp)
+				if imageInfo.Digest == "" && *imageVerify.DigestMutate && ruleResp.Status == response.RuleStatusPass {
+					err := iv.patchDigest(path, imageInfo, digest, ruleResp)
+					if err != nil {
+						ruleResp.Message = err.Error()
+						ruleResp.Status = response.RuleStatusFail
+					}
 				}
 			} else {
 				ruleResp = iv.attestImage(imageVerify, imageInfo)
+				if imageInfo.Digest == "" && *imageVerify.DigestMutate {
+					imageData, err := fetchImageDataMap(imageInfo.String())
```

Comment:
fetchImageDataMap is potentially a more expensive operation than what needs to be for calculating the digest. I would prefer if we used something else. The GGCR code that it uses should be usable for just calculating the digest. It also downloads the config while you only need the manifest to calculate the digest. Separately, you should also directly be able to use the `resolvedImage` variable inside the call to construct the correct output.

Reply:
addressed